packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -260,67 +270,81 @@ class HotRunner extends ResidentRunner {
       if (result != 0)
         return false;
     }
-    final Status devFSStatus = logger.startProgress('Syncing files to device...',
-        expectSlowOperation: true);
-    int bytes = 0;
-    try {
-      bytes = await _devFS.update(progressReporter: progressReporter,
+
+    for (RunningDevice device in runningDevices) {
+      final Status devFSStatus = logger.startProgress(
```

Comment:
same refactor here, move this code into a function that receives a RunningDevice -- better yet, can this be a function on RunningDevice?